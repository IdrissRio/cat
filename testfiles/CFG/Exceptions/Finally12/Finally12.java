public class Finally12 {
  void m(boolean bool) {
    do {
      int a;
      try {
        int b;
        try {
          int c;
          if (bool)
            break;
        } finally {
          int d;
        }
        int e;
      } finally {
        int f;
      }
      int g;
    } while (bool);
    int h;
  }
}