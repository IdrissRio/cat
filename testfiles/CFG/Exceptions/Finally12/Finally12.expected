NodeName: Entry[15] Successors: [Exit[16]]
NodeName: Exit[16] Successors: []
NodeName: Entry[7] Successors: [Exit[8]]
NodeName: Exit[8] Successors: []
NodeName: Entry[29] Successors: [VariableDeclarator[50]]
NodeName: VariableDeclarator[50] Successors: [VariableDeclarator[61]]
NodeName: VariableDeclarator[61] Successors: [VariableDeclarator[72]]
NodeName: VariableDeclarator[72] Successors: [VarAccess[76]]
NodeName: VarAccess[76] Successors: [UncheckedExceptions[78], UncheckedExceptions[118]]
NodeName: UncheckedExceptions[78] Successors: [VariableDeclarator[89]]
NodeName: VariableDeclarator[89] Successors: [VariableDeclarator[101]]
NodeName: VariableDeclarator[101] Successors: [VariableDeclarator[154], Exit[40]]
NodeName: VariableDeclarator[154] Successors: [Exit[40]]
NodeName: Exit[40] Successors: []
NodeName: UncheckedExceptions[118] Successors: [VariableDeclarator[115]]
NodeName: VariableDeclarator[115] Successors: [VariableDeclarator[124], VariableDeclarator[136]]
NodeName: VariableDeclarator[124] Successors: [UncheckedExceptions[139]]
NodeName: UncheckedExceptions[139] Successors: [VariableDeclarator[136]]
NodeName: VariableDeclarator[136] Successors: [VariableDeclarator[145], Exit[40]]
NodeName: VariableDeclarator[145] Successors: [VarAccess[148]]
NodeName: VarAccess[148] Successors: [VariableDeclarator[50], VariableDeclarator[154]]

NodeName: Exit[16] Predecessor: [Entry[15]]
NodeName: Entry[15] Predecessor: []
NodeName: Exit[8] Predecessor: [Entry[7]]
NodeName: Entry[7] Predecessor: []
NodeName: Exit[40] Predecessor: [VariableDeclarator[101], VariableDeclarator[136], VariableDeclarator[154]]
NodeName: VariableDeclarator[101] Predecessor: [VariableDeclarator[89]]
NodeName: VariableDeclarator[89] Predecessor: [UncheckedExceptions[78]]
NodeName: UncheckedExceptions[78] Predecessor: [VarAccess[76]]
NodeName: VarAccess[76] Predecessor: [VariableDeclarator[72]]
NodeName: VariableDeclarator[72] Predecessor: [VariableDeclarator[61]]
NodeName: VariableDeclarator[61] Predecessor: [VariableDeclarator[50]]
NodeName: VariableDeclarator[50] Predecessor: [Entry[29], VarAccess[148]]
NodeName: Entry[29] Predecessor: []
NodeName: VarAccess[148] Predecessor: [VariableDeclarator[145]]
NodeName: VariableDeclarator[145] Predecessor: [VariableDeclarator[136]]
NodeName: VariableDeclarator[136] Predecessor: [UncheckedExceptions[139], VariableDeclarator[115]]
NodeName: UncheckedExceptions[139] Predecessor: [VariableDeclarator[124]]
NodeName: VariableDeclarator[124] Predecessor: [VariableDeclarator[115]]
NodeName: VariableDeclarator[115] Predecessor: [UncheckedExceptions[118]]
NodeName: UncheckedExceptions[118] Predecessor: [VarAccess[76]]
NodeName: VariableDeclarator[154] Predecessor: [VariableDeclarator[101], VarAccess[148]]
